xbmc/addons/AddonManager.cpp

```diff
@@ -608,6 +610,11 @@ bool CAddonMgr::FindAddon(const std::string& addonId, const AddonVersion& addonV
   CLog::Log(LOGINFO, "CAddonMgr::{}: {} v{} installed", __FUNCTION__, addonId,
             addonVersion.asString());
 
+  if (!IsAddonInstalled(addonId))
+  {
+    m_database.AddInstalledAddon(it->second, origin);
```

Comment:
@howie-f Well, this seems not to work as expected as it results in an error:

```log
2020-10-15 22:19:14.479051+0200 kodi.bin[53060:1434162] 2020-10-15 22:19:14.479 T:1434162    INFO <general>: CAddonMgr::FindAddon: script.module.slyguy v0.12.1 installed
2020-10-15 22:19:14.479406+0200 kodi.bin[53060:1434162] 2020-10-15 22:19:14.479 T:1434162   ERROR <general>: SQL: [Addons33.db] SQLite error SQLITE_CONSTRAINT_UNIQUE (UNIQUE constraint failed: installed.addonID)
                                                   Query: INSERT INTO installed(addonID, enabled, installDate, origin) VALUES('script.module.slyguy', 1, '2020-10-15 22:19:14', 'repository.slyguy') (UNIQUE constraint failed: installed.addonID)
2020-10-15 22:19:14.479779+0200 kodi.bin[53060:1434162] 2020-10-15 22:19:14.479 T:1434162   ERROR <general>: AddInstalledAddon failed
2020-10-15 22:19:14.709867+0200 kodi.bin[53060:1433942] 2020-10-15 22:19:14.709 T:1433942    INFO <general>: CPythonInvoker(5, /Users/kai/Library/Application Support/Kodi/addons/script.module.slyguy/service.py): script successfully run
2020-10-15 22:19:14.714634+0200 kodi.bin[53060:1433950] 2020-10-15 22:19:14.714 T:1433950    INFO <general>: CAddonMgr::FindAddon: metadata.common.themoviedb.org v3.2.4 installed
2020-10-15 22:19:14.714999+0200 kodi.bin[53060:1433950] 2020-10-15 22:19:14.714 T:1433950   ERROR <general>: SQL: [Addons33.db] SQLite error SQLITE_CONSTRAINT_UNIQUE (UNIQUE constraint failed: installed.addonID)
                                                   Query: INSERT INTO installed(addonID, enabled, installDate, origin) VALUES('metadata.common.themoviedb.org', 1, '2020-10-15 22:19:14', 'repository.xbmc.org') (UNIQUE constraint failed: installed.addonID)
2020-10-15 22:19:14.715152+0200 kodi.bin[53060:1433950] 2020-10-15 22:19:14.715 T:1433950   ERROR <general>: AddInstalledAddon failed
```

Reply:
at a first glance it looks like we should check `IsAddonInstalled(addon, origin)` but i‘ll look at it in detail this afternoon

Reply:
Not, but maybe the SQL should changed from INSERT to REPLACE? Not an SQL expert, though.

Reply:
... and the method should be renamed to UpdateInstalledAddon, then?

Reply:
this would fix it but the dataset seems to there while it shouldn‘t really 🤔
as a workaround `REPLACE` would be fine